scripts/encodings_retrival/extract_scGPT_encodings.py

```diff
@@ -0,0 +1,235 @@
+# Script to extract the encodings for all genes from a scGPT model and save them to a csv file.
+# It is based on the tutorial in https://github.com/bowang-lab/scGPT/blob/main/tutorials/Tutorial_GRN.ipynb
+# The human model is taken from: https://drive.google.com/drive/folders/1oWh_-ZRdhtoGQ2Fw24HP41FgLoomVo-y
+# the blood model is taken from: https://drive.google.com/drive/folders/1kkug5C7NjvXIwQGGaGoqXTk_Lb_pDrBU
+# to run, this requires that the scgpt library will be in place.
+# requires python '<3.11,>=3.7.12'
+#   > git clone https://github.com/bowang-lab/scGPT.git
+#   > pip install -e .
+#   > pip install click
+#   > pip install gdown
+import json
+import tempfile
+from pathlib import Path
+
+import click
+import gdown
+import numpy as np
+import pandas as pd
+import torch
+from scgpt.model import TransformerModel
+from scgpt.tokenizer.gene_tokenizer import GeneVocab
+
+HUMAN_MODEL_URLS = {
+    "args.json": "https://drive.google.com/file/d/1hh2zGKyWAx3DyovD30GStZ3QlzmSqdk1/view?usp=drive_link",
+    "vocab.json": "https://drive.google.com/file/d/1H3E_MJ-Dl36AQV6jLbna2EdvgPaqvqcC/view?usp=drive_link",
+    "best_model.pt": "https://drive.google.com/file/d/14AebJfGOUF047Eg40hk57HCtrb0fyDTm/view?usp=drive_link",
+}
+
+BLOOD_MODEL_URLS = {
+    "args.json": "https://drive.google.com/file/d/1y4UJVflGl-b2qm-fvpxIoQ3XcC2umjj0/view?usp=drive_link",
+    "vocab.json": "https://drive.google.com/file/d/127FdcUyY1EM7rQfAS0YI4ms6LwjmnT9J/view?usp=drive_link",
+    "best_model.pt": "https://drive.google.com/file/d/1MJaavaG0ZZkC_yPO4giGRnuCe3F1zt30/view?usp=drive_link",
+}
+
+
+def load_url_dict(model_type):
+    """
+    Depending on the model type, load the dictionary with the correct url links.
+
+    Args:
+    ----
+        model_type (str): "blood" or "human"
+
+    Returns:
+    -------
+        a dictionary of file urls
+
+    """
+    if model_type == "human":
```

Comment:
Maybe we want to be noncase sensitive here? consider something along the lines of `model_type .lower()=='human'`